### PR TITLE
[Release v1.22.1] Prepare for snowpark python release 1.22.1

### PR DESCRIPTION
**Pull request by sfc-gh-yzou**
1) update changelog for 1.22.1
2) update versions.py
3) update meta.yaml

### CODE REVIEW

**Review by sfc-gh-jrose — APPROVED** [review]

**Review by sfc-gh-aalam — APPROVED** [review]

**Review by sfc-gh-jdu — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Release History
 
+## 1.22.1 (2024-09-11)
+This is a re-release of 1.22.0, please refer to the release note for 1.22.0 for detailed release content.
```
Comment (sfc-gh-mabrennan) [inline CHANGELOG.md:4]: I suggest this:

This is a re-release of 1.22.0. Please refer to the 1.22.0 release notes for detailed release content.

Comment (sfc-gh-yzou) [reply CHANGELOG.md]: updated

**Review by sfc-gh-mabrennan — APPROVED** [review]
changelog looks good to me.

**Comment by sfc-gh-yzou** [comment]
pr cherry picked back to main, close this on release branch

**Pull request closed without merging**